process/smartContract/process.go

```diff
@@ -193,12 +193,15 @@ func (sc *scProcessor) GasScheduleChange(gasSchedule map[string]map[string]uint6
 	defer sc.mutGasLock.Unlock()
 
 	apiCosts := gasSchedule[core.ElrondAPICost]
-	if apiCosts == nil {
+	_, existAsyncCallStepField := apiCosts[core.AsyncCallStepField]
+	_, existAsyncCallbackGasLockField := apiCosts[core.AsyncCallbackGasLockField]
+	if apiCosts == nil || !existAsyncCallStepField || !existAsyncCallbackGasLockField {
```

Comment:
Shouldn't we first check if apiCosts is nil and only then assign the bool values and check if !existAsyncCallStepField || !existAsyncCallbackGasLockField ?

Reply:
:+1:  fixed